x-pack/plugins/fleet/public/components/agent_enrollment_flyout/hooks.tsx

```diff
@@ -64,20 +76,67 @@ export function useIsK8sPolicy(agentPolicy?: AgentPolicy) {
           : 'IS_NOT_KUBERNETES'
       );
     }
+
+    async function checkifCSP() {
+      if (!agentPolicy) {
+        setIsCSP({ status: 'IS_LOADING', cloudformationUrl: 'PlaceHolder' });
+        return;
+      }
+
+      const pkgArray = agentPolicy?.package_policies?.map((e) => isCSPPackage(e));
+
+      if (pkgArray?.includes('IS_CSP_KSPM')) {
+        setIsCSP({ status: 'IS_CSP_KSPM', cloudformationUrl: 'PlaceHolder' });
+      } else if (pkgArray?.includes('IS_CSP_CSPM')) {
+        setIsCSP({ status: 'IS_CSP_CSPM', cloudformationUrl: 'PlaceHolder' });
+      } else if (pkgArray?.includes('IS_CSP_CNVM')) {
+        setIsCSP({ status: 'IS_CSP_CNVM', cloudformationUrl: 'PlaceHolder' });
+      } else {
+        setIsCSP({ status: 'IS_NOT_CSP', cloudformationUrl: 'PlaceHolder' });
+      }
+    }
+
     checkifK8s();
+    checkifCSP();
   }, [agentPolicy]);
 
-  return { isK8s };
+  return { isK8s, isCSP };
 }
 
 const isK8sPackage = (pkg: PackagePolicy) => {
   const name = pkg.package?.name as string;
-  if (name === FLEET_CLOUD_SECURITY_POSTURE_PACKAGE) {
-    return pkg.inputs.some(
-      (input) =>
-        input.enabled && input.policy_template === FLEET_CLOUD_SECURITY_POSTURE_KSPM_POLICY_TEMPLATE
-    );
-  }
 
   return K8S_PACKAGES.has(name);
 };
+
+const isCSPPackage = (pkg: PackagePolicy) => {
```

Comment:
if you want to keep the variable name, then try  to use a Set instead of multiple conditional checks. It's cleaner and easier to read and maintain

```js
 const isCSPPackage = (pkg: PackagePolicy): boolean => {
  const packageName = pkg.package?.name as string;

  if (packageName === FLEET_CLOUD_SECURITY_POSTURE_PACKAGE) {
    const policyTemplates = new Set([
      FLEET_CLOUD_SECURITY_POSTURE_KSPM_POLICY_TEMPLATE,
      FLEET_CLOUD_SECURITY_POSTURE_CSPM_POLICY_TEMPLATE,
      FLEET_CLOUD_SECURITY_POSTURE_CNVM_POLICY_TEMPLATE,
    ]);

    return pkg.inputs.some(
      (input) =>
        input.enabled && policyTemplates.has(input.policy_template)
    );
  }

  return false;
};```